packages/primitives/src/reatomArray.ts

```diff
@@ -42,5 +40,36 @@ export const reatomArray = <T>(
           }),
         `${name}.with`,
       ),
+      push: action((ctx, ...items: T[]) => {
+        const arrCopy = ctx.get(target).slice();
+        const pushed = arrCopy.push(...items);
+        target(ctx, arrCopy);
+
+        return pushed;
+      }, `${name}.push`),
+      pop: action((ctx) => {
+        const arrCopy = ctx.get(target).slice();
+        const popped = arrCopy.pop();
+        target(ctx, arrCopy);
+
+        return popped;
+      }, `${name}.pop`),
+      shift: action((ctx) => {
+        const arrCopy = ctx.get(target).slice();
+        const shifted = arrCopy.shift();
+        target(ctx, arrCopy);
+
+        return shifted;
+      }, `${name}.shift`),
+      unshift: action((ctx, ...items: T[]) => {
+        const arrCopy = ctx.get(target).slice();
+        const unshifted = arrCopy.unshift(...items);
+        target(ctx, arrCopy);
+        
+        return unshifted;
+      }, `${name}.unshift`),      slice: action(
+        (ctx, start?: number, end?: number) => target(ctx, (state) => state.slice(start, end)),
+        `${name}.slice`,
+      ),

```

Comment:
"slice"? There is totaly no neednes for this
```suggestion
      }, `${name}.unshift`),

```